Zyantific.Zydis/Native/DecoderTypes.cs

```diff
@@ -23,6 +40,457 @@ public enum AddressWidth
+
+        public ZyanU8 L;
+
+        public ZyanU8 pp;
+
+        public ZyanU8 offset;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct DecodedInstructionRawVex
+    {
+        public ZyanU8 R;
+
+        public ZyanU8 X;
+
+        public ZyanU8 B;
+
+        public ZyanU8 m_mmmm;
+
+        public ZyanU8 W;
+
+        public ZyanU8 vvvv;
+
+        public ZyanU8 L;
+
+        public ZyanU8 pp;
+
+        public ZyanU8 offset;
+
+        public ZyanU8 size;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct DecodedInstructionRawEvex
+    {
+        public ZyanU8 R;
+
+        public ZyanU8 X;
+
+        public ZyanU8 B;
+
+        public ZyanU8 R2;
+
+        public ZyanU8 Mn;
+
+        public ZyanU8 W;
+
+        public ZyanU8 vvvv;
+
+        public ZyanU8 pp;
+
+        public ZyanU8 z;
+
+        public ZyanU8 L2;
+
+        public ZyanU8 L;
+
+        public ZyanU8 b;
+
+        public ZyanU8 V2;
+
+        public ZyanU8 aaa;
+
+        public ZyanU8 offset;
+
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct DecodedInstructionRawMvex
+    {
+        public ZyanU8 R;
+
+        public ZyanU8 X;
+
+        public ZyanU8 B;
+
+        public ZyanU8 R2;
+
+        public ZyanU8 mmmm;
+
+        public ZyanU8 W;
+
+        public ZyanU8 vvvv;
+
+        public ZyanU8 pp;
+
+        public ZyanU8 E;
+
+        public ZyanU8 SSS;
+
+        public ZyanU8 V2;
+
+        public ZyanU8 kkk;
+
+        public ZyanU8 offset;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct DecodedInstructionModRm
+    {
+        public ZyanU8 mod;
+        public ZyanU8 reg;
+        public ZyanU8 rm;
+        public ZyanU8 offset;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct DecodedInstructionRawSib
+    {
+        public ZyanU8 scale;
+        public ZyanU8 index;
+        public ZyanU8 Base;
+        public ZyanU8 offset;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct DecodedInstructionRawDisp
+    {
+        public ZyanI64 value;
+        public ZyanU8 size;
+        public ZyanU8 offset;
+    }
+
+    [StructLayout(LayoutKind.Explicit)]
+    public struct DecodedInstructionRawImmValue
+    {
+        [FieldOffset(0)]
+        public ZyanU64 U;
+
+        [FieldOffset(1)]
+        public ZyanI64 S;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct DecodedInstructionRawImm
+    {
+        public ZyanBool IsSigned;
+
+        public ZyanBool IsRelative;
+
+        public DecodedInstructionRawImmValue ImmValue;
+
+        public ZyanU8 size;
+
+        public ZyanU8 offset;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct DecodedInstructionRaw
+    {
+        public ZyanU8 PrefixCount;
+
+        //   DecodedInstructionRawPrefixes[] RawPrefixes; // this HAS to be an array because it will mess up the struct otherwise.
+        [MarshalAs(UnmanagedType.ByValArray,
```

Comment:
This won't work with the `unsafe` configuration I guess. We need conditional compilation here.